test/course_bot/essay_test.clj

```diff
@@ -524,3 +524,114 @@
     (is (= 4 (essay/calculate-essay-score [3])))
     (is (= 4 (essay/calculate-essay-score [3 3])))
     (is (= 4 (essay/calculate-essay-score [3 3 3])))))
+
+(deftest essay-reupload-test
+  (let [conf (misc/get-config "conf-example/csa-2023.edn")
+        db (tt/test-database (-> conf :db-path))
+        plagiarism-db (tt/test-plagiarsm-database (-> conf :plagiarism-path))
+
+        {talk :talk, *chat :*chat}
+        (tt/test-handler (general/start-talk db conf)
+          (essay/submit-talk db conf "essay1" plagiarism-db)
+          (essay/assignreviewers-talk db conf "essay1"))]
+
+    (tt/with-mocked-morse *chat
+      ;; Register three users
+      (doseq [[id name] [[1 "Alice"] [2 "Bob"] [3 "Charlie"]]]
+        (testing (str "Register user " name)
+          (talk id "/start")
+          (talk id name)
+          (talk id "gr1")))
+
+      ;; Test scenario 1: First-time essay submission
+      (testing "First-time essay submission"
+        (talk 1 "/essay1submit")
+        (is (answers? (tt/history *chat :user-id 1)
+              [(format (tr :essay/send-essay-text-in-one-message-:essay-name) "essay1")]))
+
+        (talk 1 "Alice's essay text")
+        (tt/match-history *chat
```

Comment:
Please, use answer? like here: https://github.com/ryukzak/course-bot/blob/40396b8791a0c70f9caf2eff6a6e18c41a46fc69/test/course_bot/quiz_test.clj#L16

It will be a much more readable tests.